src/widgets/call-panel.jsx

```diff
@@ -0,0 +1,396 @@
+        .then(stream => {
+          this.setState({localStream: stream});
+          this.localRef.current.srcObject = stream;
+          // Send call invitation.
+          this.props.onInvite(this.props.topic.name, this.props.seq, this.props.callState);
+        })
+        .catch(this.handleGetUserMediaError);
+    }
+  }
+
+  createPeerConnection() {
+    var pc = new RTCPeerConnection(CALL_WEBRTC_CONFIG);
+
+    pc.onicecandidate = this.handleICECandidateEvent;
+    pc.oniceconnectionstatechange = this.handleICEConnectionStateChangeEvent;
+    pc.onicegatheringstatechange = this.handleICEGatheringStateChangeEvent;
+    pc.onsignalingstatechange = this.handleSignalingStateChangeEvent;
+    pc.onnegotiationneeded = this.handleNegotiationNeededEvent;
+    pc.ontrack = this.handleTrackEvent;
+
+    this.setState({pc: pc});
+    console.log('Created RTCPeerConnnection');
+    return pc;
+  }
+
+  handleVideoAnswerMsg(info) {
+    console.log("*** Call recipient has accepted our call");
+
+    // Configure the remote description, which is the SDP payload
+    // in our "video-answer" message.
+
+    var desc = new RTCSessionDescription(info.payload);
+    this.state.pc.setRemoteDescription(desc).catch(this.reportError);
+  }
+
+  reportError(errMessage) {
+    console.log(`Error ${errMessage.name}: ${errMessage.message}`);
+  }
+
+  handleNegotiationNeededEvent() {
+    this.state.pc.createOffer().then(offer => {
+      return this.state.pc.setLocalDescription(offer);
+    })
+    .then(() => {
+      this.props.onSendOffer(this.props.topic.name, this.props.seq, this.state.pc.localDescription.toJSON());
+    })
+    .catch(this.reportError);
+  }
+
+  handleICECandidateEvent(event) {
+    if (event.candidate) {
+      this.props.onIceCandidate(this.props.topic.name, this.props.seq, event.candidate.toJSON());
+    }
+  }
+
+  handleNewICECandidateMsg(info) {
+    var candidate = new RTCIceCandidate(info.payload);
+
+    this.state.pc.addIceCandidate(candidate)
+      .catch(this.reportError);
+  }
+
+  handleICEConnectionStateChangeEvent(event) {
+    switch (this.state.pc.iceConnectionState) {
+      case "closed":
+      case "failed":
+        this.handleCloseClick();
+        break;
+    }
+  }
+
+  handleSignalingStateChangeEvent(event) {
+    switch (this.state.pc.signalingState) {
+      case "closed":
+        this.handleCloseClick();
+        break;
+    }
+  }
+
+  handleICEGatheringStateChangeEvent(event) {
+    console.log('ICE gathering change state: ', event);
+  }
+
+  handleTrackEvent(event) {
+    this.remoteRef.current.srcObject = event.streams[0];
+    // Make sure we redraw the UI properly.
+    this.forceUpdate();
+  }
+
+  handleGetUserMediaError(e) {
+    console.log('-->', e);
+    switch(e.name) {
+      case "NotFoundError":
+        console.log("Unable to open your call because no camera and/or microphone" +
+              "were found.");
+        break;
+      case "SecurityError":
+      case "PermissionDeniedError":
```

Comment:
Please use single quotes for un-localizable strings, double quotes for localizable.